x-pack/legacy/plugins/task_manager/task_store.test.ts

```diff
@@ -36,9 +49,19 @@ const serializer = new SavedObjectsSerializer(new SavedObjectsSchema());
 beforeEach(() => jest.resetAllMocks());
 
 const mockedDate = new Date('2019-02-12T21:01:22.479Z');
+
+const anHourAgo = new Date();
+anHourAgo.setHours(anHourAgo.getHours() - 1);
+
+const anHourInTheFuture = new Date();
+anHourInTheFuture.setHours(anHourInTheFuture.getHours() + 1);
+
+const NodeDate = (global as any).Date;
+
 (global as any).Date = class Date {
-  constructor() {
-    return mockedDate;
+  constructor(date?: string) {
+    // use mock for empty Date()s, but not when wrapping a date string
+    return date ? new NodeDate(date) : mockedDate;
```

Comment:
This addresses an issue I identified in our test, where dates being returned by mocks on tasks would ignore what the test setup was setting, as all Dates were automatically becoming the `mockedDate` wven if they had a specific date specified for them.

This is dangerous as it can hide issues where dates are calculated incorrectly, but you can't tell as the code says these dates are different but under the hood they all equal each other.

I double checked none of the tests are wrong, which was lucky, but changing this prevent this problem from occurring in the future.

Reply:
Yeah in the early days I used to use mocked dates. I've learned about fake timers over time and they're much better. Feel free to implement fake timers instead if it helps https://github.com/elastic/kibana/blob/bf247018897c468ae2596f329b1fa325dc0cb523/x-pack/legacy/plugins/task_manager/task_manager.test.ts#L43-L47.

Reply:
Yeah, I felt that would make the PR bloated with an unrelated refactor- I'll likely do that later in a separate PR